Support cstruct +3.4.0

psafont opened the pull request:
Build using cstruct-lwt instead of cstruct.lwt, as per instructions on https://discuss.ocaml.org/t/psa-cstruct-3-4-0-removes-old-ocamlfind-subpackage-aliases/3275

Fixes https://github.com/mirleft/ocaml-nocrypto/issues/155

psafont committed:
Support cstruct +3.4.0

psafont commented:
Not sure what's causing the bubblewrap installation failure in the failing travis job, could be a transient failure but I can't re-trigger it:

```
+sudo apt-get install -y bubblewrap
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package bubblewrap
The command "bash -ex .travis-opam.sh" exited with 100.
```

XVilka commented on opam:
```diff
@@ -26,7 +26,7 @@ depends: [
   "ocb-stubblr" {build}
   "ounit" {test}
-  "cstruct" {>="3.0.0" & <"3.4.0"}
+  "cstruct" {>="3.0.0"}
```
Maybe make it higher version then? Since the name had changed since that version.

XVilka replied:
Ah, nevermind, it was introduced in 3.0.0

copy commented:
@avsm has already submitted patches for this package to opam-repository that fix support with cstruct 3.4 and 4.0: https://github.com/ocaml/opam-repository/pull/13655 https://github.com/ocaml/opam-repository/pull/13755

avsm commented:
Ah yes thanks -- I haven't been looking at the master branch with my patches so forward-porting them would be appreciated

psafont committed:
Support cstruct 4.0.0

Need to explicitly depend on sexplib rather than hope
it is a transitive dependency of cstruct. This lets
cstruct.4.0.0 work which makes sexplib optional.

psafont commented:
I've added an explicit dependency on sexplib on the opam file too, otherwise it wouldn't work.

psafont commented:
@pqwy is there anything stopping this PR from being merged?

XVilka commented:
@avsm @hannesm shouldn't this be merged then? Also, does it support cstruct-5.0.0?

XVilka commented:
Ping?

psafont commented:
Closing as I've replaced nocrypto with mirage-crypto on the projects I'm involved with.
